### ISSUE TITLE
Final output's audio bitrate is stuck at 130kbps

### ISSUE THREAD

**Issue by eplipswich** [rehdi93/BlenderRenderController#7]
This means that no matter what audio codec I use and what bitrate I choose, the final output video's audio bitrate is always fixed at around 130kbps for some reason. This is a big issue as I would like the final output after concatenation to have a higher audio bitrate. There are no problems with the chunks themselves (the chunks show the correct audio bitrate). It’s the joining of chunks where this issue occurs.

**Comment by red-rj** [comment]
A couple of things you can try:

- A different version of FFmpeg, as this is clearly FFmpeg doing something weird, BRC just tells it to copy the mixdown audio as-is.
- There is a special case if the audio is set to "None", it will use a AC3 default w/ fixed bitrate, unlikely to make a diference, but worth a shot.

**Comment by eplipswich** [comment]
Nah, I had tried manually concatenating **with the latest version of FFmpeg** (I had even chosen PCM for my audio codec), and it **worked perfectly**. With the BRC though, it still showed 130kbps **even with the latest version of FFmpeg**. And this is regardless of whether I join chunks **with or without the mixdown audio**. It definitely/clearly has something to do with the coding in BRC at the joining chunks part and not FFmpeg itself, and I highly recommend you look into it if you have the time.

On a side note, I also tried rendering in .avi, and instead of the audio codec being PCM like I chose in Blender, it became MP3 in 128kbps instead. Manually concatenating with FFmpeg though has no issues as usual.

**Comment by red-rj** [comment]
All that BRC does in this respect is call FFmpeg to join the chunks together.

The command sent to FFmpeg looks like this:
```
ffmpeg -f concat -safe 0 -i "path\to\chunklist.txt" -i "path\to\your_proj.a" -map 0:v -map 1:a -c:v copy -t XX:YY:ZZ "path\to\your_proj.v" -y
```
Where "chunklist.txt" is a text file containing the paths of all chunk files, and "your_proj.a" is the final audio. Whats the bitrate of this audio file? (It should be in the output folder, alongside the final vid)

**Comment by eplipswich** [comment]
The audio file is FLAC codec, so it's lossless/no bitrate (well, it's supposed to be PCM, but I guess it doesn't matter). However, the final output video still shows 130kbps (and even worse: AAC). The video that I just rendered with BRC is a 6 second video in Quicktime container, H.264 video codec and PCM audio codec.

I think the problem probably lies somewhere in _-map 0:v -map 1:a -c:v copy -t XX:YY:ZZ "path\to\your_proj.v" -y_.  Because in my code (for manual concatenation), all I used was 

_ffmpeg -f concat -safe 0 -i chunklist.txt -c copy output.mov_ 

And this line of code works perfect (of course, this doesn't generate the mixdown audio, but it's not a problem for me since I render long videos). Maybe if you just write it as 

_ffmpeg -f concat -safe 0 -i chunklist.txt -i "path\to\your_proj.a" -map 0:v -map 1:a -c copy -t XX:YY:ZZ "path\to\your_proj.v"_

it'll work?

**Comment by red-rj** [comment]
`-map 0:v -map 1:a -c:v copy` just maps the mixdown audio to the joined video and `-t XX:YY:ZZ` is the total runtime. Kinda at a loss right now sadly ☹️ 

Also you CAN skip the mixdown audio altogether in BRC `In the main window: Joining action -> Join chunks`. Just make sure your audio is ok and in sync, as that can sometimes happend

**Comment by eplipswich** [comment]
Yes, I know about the skip mixdown audio thing. The problem is, this bug happens even WITHOUT the mixdown audio (I mentioned that in my 2nd post).

Maybe I’ll test out that line of code of yours manually later and see how it goes. Can you try testing it out manually too (just that line of code).

**Comment by red-rj** [comment]
My theory so far is that FFmpeg is assuming some odd defaults due to the arguments BRC is sending

`-c:v copy` copies the video, and I guess `-c copy` copies both video and audio (not sure), but BRC passes the former on both cases

**Comment by eplipswich** [comment]
Yep, you're right. I tested both **-c:v copy** and **-c cop**y and it turns out that **-c copy** is the one that works perfectly (my guess is that -c:v copy only copies the video's codec, so it randomly generates its own codec for the audio since there's no specification for the audio). 

So if you can, may I trouble you to change **-c:v copy** to **-c copy**? Many thanks!

**Comment by red-rj** [comment]
6bd0349 should be good now
could you [test it?](https://github.com/RedRaptor93/BlenderRenderController/releases/tag/1.2.0.1_hotfix)

**Comment by eplipswich** [comment]
Concatenating without mixdown audio works perfect, but concatenating **with** mixdown audio still has the bug. I think you forgot to change the code for **with** mixdown audio too.

**Comment by red-rj** [comment]
I know, I wanted to make sure that part got fixed, should have a full fix out by the weekend

**Comment by red-rj** [comment]
fixed in 0930bfa (release [v1.2.0.1](https://github.com/RedRaptor93/BlenderRenderController/releases/tag/v1.2.0.1))
